docs/cookbook/custom-promotion-rule.rst

```diff
@@ -0,0 +1,115 @@
+How to add a custom promotion rule?
+===================================
+
+Adding new, custom rules to your shop is a common usecase. You can imagine for instance, that you have some customers
+in your shop that you distinguish as premium. And for these premium customers you would like to give special promotions.
+For that you will need a new PromotionRule that will check if the customer is premium or not.
+
+Create a new promotion rule
+---------------------------
+
+The new Rule needs a RuleChecker class:
+
+.. code-block:: php
+
+    <?php
+
+    namespace AppBundle\Promotion\Checker\Rule;
+
+    use Sylius\Component\Promotion\Checker\Rule\RuleCheckerInterface;
+    use Sylius\Component\Promotion\Model\PromotionSubjectInterface;
+
+
+    class PremiumCustomerRuleChecker implements RuleCheckerInterface
+    {
+        const TYPE = 'premium_customer';
+
+        /**
+         * {@inheritdoc}
+         */
+        public function isEligible(PromotionSubjectInterface $subject, array $configuration)
+        {
+            return $subject->getCustomer()->isPremium();
+        }
+
+        /**
+         * {@inheritdoc}
+         */
+        public function getConfigurationFormType()
+        {
+            return 'app_promotion_rule_premium_customer_configuration';
+        }
+    }
+
+Register the new rule checker as a service in the ``AppBundle/Resources/config/services.yml``:
+
+.. code-block:: yaml
+
+    # AppBundle/Resources/config/services.yml
+    app.promotion_rule_checker.premium_customer:
+        class: AppBundle\Promotion\Checker\Rule\PremiumCustomerRuleChecker
+        tags:
+            - { name: sylius.promotion_rule_checker, type: premium_customer, label: Premium customer }
+
+Prepare a configuration form type for your new rule
+---------------------------------------------------
+
+To be able to configure a promotion with your new rule you will need a form type fir the admin panel.
+
+Create the form type class:
+
+.. code-block:: php
+
+    <?php
+
+    namespace AppBundle\Form\Type\Rule;
+
+    use Symfony\Component\Form\AbstractType;
+    use Symfony\Component\Form\FormBuilderInterface;
+    use Symfony\Component\Validator\Constraints\NotBlank;
+    use Symfony\Component\Validator\Constraints\Type;
+
+    class PremiumCustomerConfigurationType extends AbstractType
+    {
+        /**
+         * {@inheritdoc}
+         */
+        public function buildForm(FormBuilderInterface $builder, array $options)
+        {
+            $builder
+                ->add('amount', 'sylius_money', [
+                    'label' => 'app.form.promotion_rule.premium_customer_configuration.amount',
```

Comment:
Do we need `amount` in this configuration? We do not use this value anywhere. IMO this rule shouldn't have any configuration fields.